bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/TemplateListSelectionPage.java

```diff
@@ -143,5 +207,9 @@ public void setVisible(boolean visible)
 		}
 		super.setVisible(visible);
 	}
-
+	
+	public void setMainPage(WizardNewProjectCreationPage mainPage) 
+	{
+		this.mainPage = mainPage;
+	}
```

Comment:
We can avoid this method if we use getPreviousPage() and access the required methods/calls